Source/WebKit/UIProcess/API/C/playstation/WKPagePrivatePlayStation.cpp

```diff
@@ -172,10 +176,15 @@ void WKPagePaint(WKPageRef pageRef, unsigned char* surfaceData, WKSize wkSurface
     auto page = WebKit::toImpl(pageRef);
     auto& backgroundColor = page->backgroundColor();
     page->endPrinting();
-    if (auto* drawingArea = static_cast<WebKit::DrawingAreaProxyCoordinatedGraphics*>(page->drawingArea())) {
+    if (auto* drawingArea = page->drawingArea()) {
         // FIXME: We should port WebKit1's rect coalescing logic here.
         WebCore::Region unpaintedRegion;
-        drawingArea->paint(ctx, paintRect, unpaintedRegion);
+#if USE(GRAPHICS_LAYER_WC)
+        if (is<WebKit::DrawingAreaProxyWC>(drawingArea))
```

Comment:
You can't switch DrawingAreaProxyWC and DrawingAreaProxyCoordinatedGraphics runtime.
The compile time switch by "#if USE(GRAPHICS_LAYER_WC)" is enough.

Reply:
Both implementations of `PageClientImpl::createDrawingAreaProxy` will return and instance of `DrawingAreaProxyWC` if GPU process is enabled and an instance of `DrawingAreaProxyCoordinatedGraphics` if its not.

https://github.com/WebKit/WebKit/blob/02cdac5e408458239c489b1a2c57a3b858f9041c/Source/WebKit/UIProcess/win/PageClientImpl.cpp#L55-L62

https://github.com/WebKit/WebKit/blob/02cdac5e408458239c489b1a2c57a3b858f9041c/Source/WebKit/UIProcess/playstation/PageClientImpl.cpp#L51-L60

Reply:
I created GraphicsLayerWC to be switchable with GraphicsLayerTextureMapper to support both GPU process mode and non-GPU process mode.
GraphicsLayerWC can't be switchable with CoordinatedGraphicsLayer which is used by PlayStation port's non-GPU process mode.
If Windows WebKit Playwight swiches from GraphicsLayerTextureMapper to GraphicsLayerWC, we can modify GraphicsLayerWC to be switchable with CoordinatedGraphicsLayer for PlayStation port's non-GPU process mode.